### DISCUSSION TITLE
CiteprocParseError: Multiple layout elements present in bibliography

### DISCUSSION THREAD

**Discussion by Dagwbl** [quarto-dev/quarto-cli#4598] [Q&A]
When I applied a CSL file, this error appeared and could not be rendered. I was confused

`CiteprocParseError: Multiple layout elements present in bibliography`

the csl file link:

[410shanghai-jiao-tong-university.csl](https://github.com/redleafnew/Chinese-STD-GB-T-7714-related-csl/blob/main/410shanghai-jiao-tong-university.csl)

**Comment by cscheid** [comment]
Thanks for the feedback. Unfortunately, we need a full reproducible example (.qmd, render command, quarto version, system details) to be able to track your issue.

**Reply by Dagwbl** [reply]
OK, I will send you the relevant documents at 9: 00 a.m. Beijing time.

**Comment by rgaiacs** [comment]
Dear @Dagwbl, where did you get the `csl` file?

Starting in [line 502](https://github.com/redleafnew/Chinese-STD-GB-T-7714-related-csl/blob/main/410shanghai-jiao-tong-university.csl#L502), you have

```
  <citation collapse="citation-number" after-collapse-delimiter=",">
    <sort>
      <key variable="citation-number"/>
    </sort>
    <layout vertical-align="sup" delimiter="," prefix="[" suffix="]">
      <text variable="citation-number"/>
      <group prefix="(" suffix=")">
        <!-- 国标要求在方括号外著录引文页码，但是跟同处引用多篇文献的情况冲突，在 CSL 中无法实现 -->
        <label variable="locator" suffix=". " form="short" strip-periods="true"/>
        <text variable="locator"/>
      </group>
    </layout>
  </citation>
  <bibliography entry-spacing="0" et-al-min="4" et-al-use-first="3" second-field-align="flush">
    <layout locale="en">
      <text variable="citation-number" prefix="[" suffix="]"/>
      <text macro="entry-layout"/>
    </layout>
    <layout>
      <text variable="citation-number" prefix="[" suffix="]"/>
      <text macro="entry-layout"/>
    </layout>
  </bibliography>
```

`csl` file can have a single `<layout>` element but the snipet above has three.

**Reply by Dagwbl** [reply]
I got it from this repo:

https://github.com/redleafnew/Chinese-STD-GB-T-7714-related-csl

As far as I know, this is the extended style for csl

You can refer here：

https://juris-m.github.io/styles/

**Reply by Dagwbl** [reply]
csl-m docs：

https://citeproc-js.readthedocs.io/en/latest/csl-m/index.html

**Reply by rgaiacs** [reply]
Pandoc does **not** have support to CSL-M. You can ask for the support at https://github.com/jgm/pandoc.

**Reply by cscheid** [reply]
@Dagwbl those docs you cite say:

> Extensions add entirely new elements or attributes to the schema. Styles making use of extended syntax will fail validation under the official CSL schema, and can be used only with Multilingual Zotero.

**Reply by Dagwbl** [reply]
> Pandoc does **not** have support to CSL-M. You can ask for the support at https://github.com/jgm/pandoc.

Thank you, I think that's the reason, I'll try it

**Reply by Dagwbl** [reply]
> @Dagwbl those docs you cite say:
> 
> > Extensions add entirely new elements or attributes to the schema. Styles making use of extended syntax will fail validation under the official CSL schema, and can be used only with Multilingual Zotero.

You are right. but I cite so much references in quarto now. I don't know how to convert to zotero.

The rendered docx doesn't have the meta citation information

**Comment by a76854** [comment]
兄弟，这个问题解决了吗，我在用pandoc转换latex，也出了这个问题。

**Reply by mcanouil** [reply]
@a76854 please post in English for us to be able to help you. Thank you.

**Reply by a76854** [reply]
I have a same error when I use pandoc to convert LaTeX to docx with specified [CSL file](https://github.com/redleafnew/Chinese-STD-GB-T-7714-related-csl/blob/main/003gb-t-7714-2015-numeric-bilingual-no-url-doi.csl), according to what I got from net, it seems that pandoc don't support some CSL-M, I want to know if this guy solved it. Thanks for you concern, my dude.

**Reply by cderv** [reply]
This is the initial discussion in Pandoc Citeproc that lead to the addition of an error being thrown by Pandoc  when several `layout` (https://github.com/jgm/citeproc/commit/cb54223919ecd327250f1b167e4e0c61473f402e)
* https://github.com/jgm/citeproc/issues/120

You could chime in there, or open a new in https://github.com/jgm/pandoc repo
